File: app/src/main/java/tpcreative/co/qrscanner/common/entities/SaveEntity.java
package tpcreative.co.qrscanner.common.entities;
import androidx.room.Entity;
import androidx.room.PrimaryKey;
import com.google.zxing.BarcodeFormat;
import java.io.Serializable;
import tpcreative.co.qrscanner.common.Utils;
import tpcreative.co.qrscanner.model.SaveEntityModel;

@Entity(tableName = "save")
public class SaveEntity implements Serializable{
    @PrimaryKey(autoGenerate = true)
    public int id;
    public String email;
    public String subject ;
    public String message;
    public String phone;
    public double lat;
    public double lon;
    public String query;
    public String title;
    public String location;
    public String description;
    public String startEvent;
    public String endEvent;
    public long   startEventMilliseconds;
    public long   endEventMilliseconds;
    public String fullName;
    public String address;
    public String text;
    public String ssId;
    public boolean hidden;
    public String password;
    public String url;
    public String createType;
    public String networkEncryption;
    public String createDatetime;
    public String barcodeFormat;
    public boolean favorite;
    public String updatedDateTime;

    /*content_type_barcode*/
    public String contentUnique;
    /*sync data*/
    public boolean isSynced;
    public String uuId;

    public SaveEntity(
                String email,
                String subject ,
                String message,
                String phone,
                double lat,
                double lon,
                String query,
                String title,
                String location,
                String description,
                String startEvent,
                String endEvent,
                long startEventMilliseconds,
                long endEventMilliseconds,
                String fullName,
                String address,
                String text,
                String ssId,
                boolean hidden,
                String password,
                String url,
                String createType,
                String networkEncryption,
                String createDatetime,BarcodeFormat barcodeFormat,String updatedDateTime){

        this.email = email;
        this.subject = subject;
        this.message = message ;
        this.phone = phone;
        this.lat = lat;
        this.lon = lon;
        this.query = query;
        this.title = title;
        this.location = location;
        this.description = description;
        this.startEvent = startEvent;
        this.startEventMilliseconds  = startEventMilliseconds;
        this.endEventMilliseconds = endEventMilliseconds;
        this.endEvent = endEvent;
        this.fullName = fullName;
        this.address = address;
        this.text = text;
        this.ssId = ssId;
        this.hidden = hidden;
        this.password = password;
        this.url = url;
        this.createType = createType;
        this.networkEncryption = networkEncryption;
        this.createDatetime = createDatetime;
        this.barcodeFormat = barcodeFormat.name();
        this.favorite = false;
        this.updatedDateTime = updatedDateTime;
    }

    public SaveEntity(){
        this.email = "";
        this.subject = "";
        this.message = "" ;
        this.phone = "";
        this.lat = 0;
        this.lon = 0;
        this.query = "";
        this.title = "";
        this.location = "";
        this.description = "";
        this.startEvent = "";
        this.endEvent = "";
        this.startEventMilliseconds = 0;
        this.endEventMilliseconds = 0;
        this.fullName = "";
        this.address = "";
        this.text = "";
        this.ssId = "";
        this.hidden = false;
        this.password = "";
        this.url = "";
        this.createType = "";
        this.networkEncryption = "";
        this.barcodeFormat = BarcodeFormat.QR_CODE.name();
        this.favorite = false;
        this.updatedDateTime = Utils.getCurrentDateTimeSort();
    }

    public SaveEntity(SaveEntityModel item){
        this.id = item.id;
        this.email = item.email;
        this.subject  = item.subject;
        this.message = item.message;
        this.phone = item.phone;
        this.lat = item.lat;
        this.lon = item.lon;
        this.query = item.query;
        this.title = item.title;
        this.location = item.location;
        this.description = item.description;
        this.startEvent = item.startEvent;
        this.endEvent = item.endEvent;
        this.startEventMilliseconds = item.startEventMilliseconds;
        this.endEventMilliseconds = item.endEventMilliseconds;
        this.fullName = item.fullName;
        this.address = item.address;
        this.text = item.text;
        this.ssId = item.ssId;
        this.hidden = item.hidden;
        this.password = item.password;
        this.url = item.url;
        this.createType = item.createType;
        this.networkEncryption = item.networkEncryption;
        this.createDatetime = item.createDatetime;
        this.barcodeFormat = item.barcodeFormat;
        this.favorite = item.favorite;
        this.updatedDateTime = item.updatedDateTime;
        this.contentUnique = item.contentUnique;
        this.isSynced = item.isSynced;
        this.uuId = item.uuId;
    }
}

File: app/src/main/java/tpcreative/co/qrscanner/common/entities/SaveEntity.kt
package tpcreative.co.qrscanner.common.entities
import androidx.room.Entity
import androidx.room.PrimaryKey
import com.google.zxing.BarcodeFormat
import tpcreative.co.qrscanner.common.Utils
import tpcreative.co.qrscanner.model.SaveEntityModel
import java.io.Serializable

@Entity(tableName = "save")
class SaveEntity : Serializable {
    @PrimaryKey(autoGenerate = true)
    var id  = 0
    var email: String?
    var subject: String?
    var message: String?
    var phone: String?
    var lat: Double
    var lon: Double
    var query: String?
    var title: String?
    var location: String?
    var description: String?
    var startEvent: String?
    var endEvent: String?
    var startEventMilliseconds: Long
    var endEventMilliseconds: Long
    var fullName: String?
    var address: String?
    var text: String?
    var ssId: String?
    var hidden: Boolean
    var password: String?
    var url: String?
    var createType: String?
    var networkEncryption: String?
    var createDatetime: String? = ""
    var barcodeFormat: String?
    var favorite: Boolean
    var updatedDateTime: String?

    /*content_type_barcode*/
    var contentUnique: String? = ""

    /*sync data*/
    var isSynced : Boolean = false
    var uuId: String? = ""
    var noted : String? = ""

    constructor() {
        email = ""
        subject = ""
        message = ""
        phone = ""
        lat = 0.0
        lon = 0.0
        query = ""
        title = ""
        location = ""
        description = ""
        startEvent = ""
        endEvent = ""
        startEventMilliseconds = 0
        endEventMilliseconds = 0
        fullName = ""
        address = ""
        text = ""
        ssId = ""
        hidden = false
        password = ""
        url = ""
        createType = ""
        networkEncryption = ""
        barcodeFormat = BarcodeFormat.QR_CODE.name
        favorite = false
        updatedDateTime = Utils.getCurrentDateTimeSort() ?: ""
        noted  = ""
    }

    constructor(item: SaveEntityModel?) {
        id = item?.id ?: 0
        email = item?.email ?: ""
        subject = item?.subject ?: ""
        message = item?.message ?: ""
        phone = item?.phone ?: ""
        lat = item?.lat ?: 0.0
        lon = item?.lon ?: 0.0
        query = item?.query ?: ""
        title = item?.title ?: ""
        location = item?.location ?: ""
        description = item?.description ?: ""
        startEvent = item?.startEvent ?: ""
        endEvent = item?.endEvent ?: ""
        startEventMilliseconds = item?.startEventMilliseconds ?: 0
        endEventMilliseconds = item?.endEventMilliseconds ?: 0
        fullName = item?.fullName ?: ""
        address = item?.address ?: ""
        text = item?.text ?: ""
        ssId = item?.ssId ?: ""
        hidden = item?.hidden ?: false
        password = item?.password  ?: ""
        url = item?.url ?: ""
        createType = item?.createType ?: ""
        networkEncryption = item?.networkEncryption ?: ""
        createDatetime = item?.createDatetime ?: ""
        barcodeFormat = item?.barcodeFormat ?: ""
        favorite = item?.favorite ?: false
        updatedDateTime = item?.updatedDateTime ?: ""
        contentUnique = item?.contentUnique ?: ""
        isSynced = item?.isSynced ?: false
        uuId = item?.uuId ?: ""
        noted = item?.noted ?: ""
    }
}